src/map/map.cpp

```diff
@@ -2651,18 +2651,18 @@ bool map_addnpc(int16 m,struct npc_data *nd)
 /*==========================================
  * Add an instance map
  *------------------------------------------*/
-int map_addinstancemap(const char *name, unsigned short instance_id)
+int map_addinstancemap(int src_m, unsigned short instance_id)
 {
-	int16 src_m = map_mapname2mapid(name);
-	char iname[MAP_NAME_LENGTH];
+	const char *iname;
```

Comment:
Since, you are refactoring this code. You should declare this variables at the moment that they are being use.